graph/patterns/default-properties.md

```diff
@@ -0,0 +1,201 @@
+  - the `@microsoft.graph.tips` instance annotation value **must** contain "This request only returns a subset of the resource's properties. Your app will need to use $select to return non-default properties. To find out what other properties are available for this resource see https://learn.microsoft.com/graph/api/resources/{entityTypeName}".
+- Callers must be able to use `$filter` with non-default properties, even though they won't show up by default in the response.
+
+Additionally, for incoming requests targeting an entity type where the caller does not specify a `$select` clause, the API Gateway Service will inject a `@microsoft.graph.tips` instance annotation, informing callers to use $select, when in "developer mode".
+API producers who use [response passthrough](https://dev.azure.com/msazure/One/_wiki/wikis/Microsoft%20Graph%20Partners/391069/Enabling-response-passthrough) must also implement this behavior, supplying the same information as shown in the [examples section below](#calling-an-api-without-using-select).
+
+## Examples
+
+In this example we'll use the following `channel` entity type.
+
+```xml
+<EntityType Name="channel" BaseType="graph.entity">
+  <Property Name="createdDateTime" Type="Edm.DateTimeOffset"/>
+  <Property Name="description" Type="Edm.String"/>
+  <Property Name="displayName" Type="Edm.String" Nullable="false"/>
+  <Property Name="email" Type="Edm.String"/>
+  <Property Name="isFavoriteByDefault" Type="Edm.Boolean"/>
+  <Property Name="membershipType" Type="graph.channelMembershipType"/>
+  <!-- moderationSettings is a new computed property that is very expensive -->
+  <Property Name="moderationSettings" Type="graph.channelModerationSettings"/> 
+  <Property Name="webUrl" Type="Edm.String"/>
+  <Property Name="filesFolderWebUrl" Type="Edm.String"/>
+</EntityType>
+```
+
+In this scenario, the API producer wants to add the `moderationSettings` property to the `channel` entity type.
+But when paging through 1000 channels at a time, this additional property will introduce a considerable increase in the response times.
+The API producer will use the default properties pattern here, and **not** return `moderationSettings` by default.
+
+### Calling an API with default properties
+
+In this example, the caller, using Graph Explorer, does not use $select, and the API returns just the default properties.
+
+#### Request
+
+```http
+GET /teams/{id}/channels
+Prefer: ms-graph-dev-mode
+```
+
+#### Response
+
+```http
+200 ok
+Content-type: application/json
+```
+
+```json
+{
+    "@odata.context": "https://graph.microsoft.com/beta/$metadata#Collection(microsoft.graph.channel)",
+    "@microsoft.graph.tips": "This request only returns a subset of the resource properties. Your app will need to use $select to return non-default properties. To find out what other properties are supported for this resource, please see the Properties section in https://learn.microsoft.com/graph/api/resources/channel.",
```

Comment:
@XVincentX because it's absolutely intended for humans and not machines :)
It's absolutely not intended for machines, because we want to inform developers so that they can take corrective action in their code.  These messages are only provided in "developer mode" - so pointless when a developer doesn't see them.

For default properties we get a **ton** of support calls from developers who don't understand why certain properties are not showing up.

Reply:
I was hoping to keep the tip as a string, but I see your point:

```json
  "@microsoft.graph.tips": {
       "code": "Default_Properties",
       "message": "This request only returns a subset of the resource properties. Your app will need to use $select to return non-default properties. To find out what other properties are supported for this resource, please see the Properties section in https://learn.microsoft.com/graph/api/resources/channel."
  }
```

Reply:
If we want to convey Graph tips via standard output in the SDK, we can definitely do that. But I don't see a need for having a code to do that?  I don't think we are going to build a mechanism where developers can choose which tips they do and do not want to have emitted to standard out.

Reply:
@darrelmiller Not sure if you've seen Vincenzo's response.

Reply:
Triage: Doc what is implemented and file a work item to change this to a collection of complex types (for improved messages and future extensibility).

Reply:
Filing a work item for this.